jax-stuff.md

```diff
@@ -77,51 +78,52 @@ _styles: >
 
 ## How Does Parallelism Work in JAX?
 
-JAX supports two schools of thought for multi-device programming:
+JAX supports three levels of parallelism for multi-device programming:
 
-1. **Compiler, take the wheel!** Let the compiler automatically partition arrays and decide what communication to add to facilitate a given program. This lets you write a program on a single device and automatically run it on hundreds without changing anything. 
-2. **Just let me write what I mean, damnit!** While compilers are nice, they sometimes do the wrong thing and add communication you don't intend. Sometimes we want to be extremely explicit about what we're doing.
+1. **Compiler-based automatic parallelization** You program as if using a single global machine, and the compiler chooses how to shard data and partition computation (with some user-provided constraints).
+2. **Explicit sharding and automatic partitioning** You still have a global view but data shardings are
+explicit in JAX types, inspectable using `jax.typeof`.
+3. **Manual per-device programming** You have a per-device view of data
```

Comment:
I think the original version of (3) is better, or some combination like "While compilers are nice, sometimes all we want is to control everything ourselves. You get to write device local code, and specify all the communication yourself."